LLama/Native/NativeLibraryConfig.cs

```diff
@@ -120,13 +129,62 @@ public NativeLibraryConfig WithLogs(bool enable = true)
             return this;
         }
 
+        /// <summary>
+        /// Add self-defined search directories. Note that the file stucture of the added 
+        /// directories must be the same as the default directory. Besides, the directory 
+        /// won't be used recursively.
+        /// </summary>
+        /// <param name="directoriesAndPriorities">The directories and corresponding priorities, in which 0 is the lowest. The default path has priority 0.</param>
+        /// <returns></returns>
+        public NativeLibraryConfig WithSearchDirectories(IDictionary<string, int> directoriesAndPriorities)
```

Comment:
Would it makes more sense to use an array for the directories (here and for `_searchDirectories`) which is implicitly in priority order?